### PR TITLE
Peru

### PR DESCRIPTION
**Pull request by Vesihiisi**
Mapping: https://www.wikidata.org/wiki/Wikidata:WikiProject_WLM/Mapping_tables/pe_(es)
Preview: https://www.wikidata.org/wiki/Wikidata:WikiProject_WLM/Mapping_tables/pe_(es)/preview

Task: https://phabricator.wikimedia.org/T176714

### CODE REVIEW

File: importer/PeEs.py [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+        if adm_q is not None:
+            self.add_statement("located_adm", adm_q)
+        else:
+            self.add_to_report("municipio", self.municipio, "located_adm")
+
+    def set_heritage_id(self):
+        wlm_prefix = self.mapping["table_name"].upper()
+        self.add_statement("wlm_id", "{}-{}".format(wlm_prefix, self.id))
+
+    def exists_with_monument_article(self, language):
```
Comment (lokal-profil) [inline importer/PeEs.py]: See https://github.com/Vesihiisi/COH-tools/pull/76#discussion_r136988915 and #126

File: importer/PeEs.py [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+        against external list.
+        If failed, use the Region iso code, which is a
+        bigger unit.
+        """
+        adm_q = None
+        municip_dic = self.data_files["municipalities"]
+        reg_dic = self.data_files["regions"]
+
+        municip_q = utils.q_from_first_wikilink("es", self.municipio)
+        if municip_q in [x["item"] for x in municip_dic]:
```
Comment (lokal-profil) [inline importer/PeEs.py]: use `utils.get_item_from_dict_by_key(dict_name=municip_dic, search_term=municip_q, search_in="item")` for consistency

File: importer/PeEs.py [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+            adm_q = municip_q
```
Comment (lokal-profil) [inline importer/PeEs.py]: Add to report already if this fails (since iso is less specific)

File: importer/PeEs.py [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+        Use the linked Municipality first, checking
+        against external list.
+        If failed, use the Region iso code, which is a
+        bigger unit.
+        """
+        adm_q = None
+        municip_dic = self.data_files["municipalities"]
+        reg_dic = self.data_files["regions"]
+
+        municip_q = utils.q_from_first_wikilink("es", self.municipio)
```
Comment (lokal-profil) [inline importer/PeEs.py]: does `self.municipio` always have a value?

File: importer/PeEs.py [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+
+class PeEs(Monument):
+
+    def set_location(self):
+        """Set the Location, using linked article."""
+        if self.has_non_empty_attribute("localidad"):
+            if utils.count_wikilinks(self.localidad) == 1:
+                loc_q = utils.q_from_first_wikilink("es", self.localidad)
+
+            if loc_q is not None:
```
Comment (lokal-profil) [inline importer/PeEs.py]: if loc_q: add_statement
else: report

File: importer/PeEs.py (hunk 4 above)
Comment (Vesihiisi) [reply importer/PeEs.py]: Yes (and it's always wikilinked).

File: importer/PeEs.py (hunk 3 above)
Comment (Vesihiisi) [reply importer/PeEs.py]: Done.

File: importer/PeEs.py (hunk 1 above)
Comment (Vesihiisi) [reply importer/PeEs.py]: Done.

File: importer/PeEs.py (hunk 2 above)
Comment (Vesihiisi) [reply importer/PeEs.py]: Done.

File: importer/PeEs.py (hunk 5 above)
Comment (Vesihiisi) [reply importer/PeEs.py]: Done.

**Comment by Vesihiisi** [comment]
Mapped direccion to directions.

**Comment by Vesihiisi** [comment]
```
Loaded offline data file: mappings/peru_regions.json
Traceback (most recent call last):
  File "PeEs.py", line 102, in <module>
    importer.main(args, dataset)
  File "/home/alicia/Dropbox/work/COH-tools/importer/importer.py", line 254, in main
    get_items(connection, dataset, upload, short, offset, table, list_matches)
  File "/home/alicia/Dropbox/work/COH-tools/importer/importer.py", line 191, in get_items
    row, mapping, data_files, existing, wikidata_site)
  File "PeEs.py", line 92, in __init__
    self.set_wd_item(self.find_matching_wikidata(mapping))
  File "/home/alicia/Dropbox/work/COH-tools/importer/Monument.py", line 531, in find_matching_wikidata
    self.mapping["language"])
  File "PeEs.py", line 71, in exists_with_monument_article
    linked="monumento", language=language, blocks="enlace")
  File "/home/alicia/Dropbox/work/COH-tools/importer/Monument.py", line 395, in get_multi_param_monument_article
    return super().exists_with_monument_article(language, linked)
AttributeError: 'super' object has no attribute 'exists_with_monument_article'
<class 'AttributeError'>
CRITICAL: Closing network session.
```

File: importer/PeEs.py (hunk 4 above)
Comment (lokal-profil) [reply importer/PeEs.py]: Thanks. updated the mapping

**Comment by lokal-profil** [comment]
There seem to be plenty of city like matches in [/pe_(es)/matches](https://www.wikidata.org/wiki/Wikidata:WikiProject_WLM/Mapping_tables/pe_(es)/matches) but looking at the name of the monuments that might actually be valid (for most of them at least)

**Comment by lokal-profil** [comment]
Preview updated. New matches running

**Comment by lokal-profil** [comment]
[Matches updated](https://www.wikidata.org/w/index.php?title=Wikidata:WikiProject_WLM/Mapping_tables/pe_(es)/matches&oldid=587935414):
* [Q11424](https://www.wikidata.org/wiki/Q11424) gets handled by #157

**Pull request merged**